### PR TITLE
#1489: Make LengthOf accept Text in ctor

### PR DESCRIPTION
**Pull request by andreoss**
Per #1489 
- Add ctor for Text

### CODE REVIEW

File: src/test/java/org/cactoos/scalar/LengthOfTest.java [hunk 1]
```diff
@@ -185,4 +186,15 @@ public void lengthOfEmptyIterator() {
             new IsEqual<>(0)
         ).affirm();
     }
+
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/scalar/LengthOfTest.java:189]: @andreoss Please, add a test when the string is empty.

File: src/test/java/org/cactoos/scalar/LengthOfTest.java [hunk 2]
```diff
@@ -185,4 +186,15 @@ public void lengthOfEmptyIterator() {
...
+    @Test
+    public void lengthOfText() {
+        new Assertion<>(
+            "Must calculate length of empty iterator",
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/scalar/LengthOfTest.java:193]: @andreoss please, fix this message, because the string isn't empty.

File: src/test/java/org/cactoos/scalar/LengthOfTest.java [hunk 3]
```diff
@@ -185,4 +186,15 @@ public void lengthOfEmptyIterator() {
...
+            new LengthOf(
+                new TextOf("abcd")
+            ).intValue(),
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/scalar/LengthOfTest.java:196]: @andreoss please, add a test to `LengthOf(...).longValue()`, `LengthOf(...).floatValue()`and `LengthOf(...).doubleValue()`

**Review by fabriciofx — CHANGES_REQUESTED** [review]
@andreoss Please, take a look.

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 3 above)
Comment (andreoss) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java]: @fabriciofx I did so, but kept it in a single test. Perhaps there could be `IsNumber` matcher for that.

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 2 above)
Comment (andreoss) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java]: @fabriciofx Fixed

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 1 above)
Comment (andreoss) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java:191]: @fabriciofx Added a test

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 3 above)
Comment (fabriciofx) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java]: @andreoss Please, can you add these tests again and open an issue in `cactoos-matchers` suggesting something like this `isNumber`?

**Review by fabriciofx — APPROVED** [review]
@andreoss It looks fine to me. @victornoel Can you merge it, please?

File: src/test/java/org/cactoos/scalar/LengthOfTest.java [hunk 4]
```diff
@@ -185,4 +188,28 @@ public void lengthOfEmptyIterator() {
...
+        new Assertion<>(
+            "Must calculate length of empty string",
+            new LengthOf(new TextOf("")).intValue(),
+            new IsEqual<>(0)
+        ).affirm();
+    }
+
+    @Test
+    public void lengthOfText() {
+        final Number num = new LengthOf(new TextOf("abcd"));
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/LengthOfTest.java:203]: @andreoss can we add a test with some utf8 characters that does not translate directly to one byte? :)

**Review by victornoel — CHANGES_REQUESTED** [review]

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 4 above)
Comment (andreoss) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java:203]: @victornoel You're suggestging that `LengthOf` should return a character-length for `Text`, not a byte-length? It seems inconsistent with how `Input` is handled.

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 3 above)
Comment (andreoss) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java]: @fabriciofx Filed. https://github.com/llorllale/cactoos-matchers/issues/207

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 4 above)
Comment (victornoel) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java:203]: @andreoss exactly, that was a sneaky way of raising this problem in your solution, sorry about that ^^ We need to thus change the design to support both cases.

One possible solution is to have two classes (maybe as private inner class for now) for `Input` and `Text` and then have `LengthOf` delegate to them.

Comment (andreoss) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java:221]: @victornoel Done. It calls `asString().length()` for `Text` now.

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 3 above)
Comment (fabriciofx) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java]: @andreoss thanks!

**Comment by fabriciofx** [comment]
@andreoss It seems fine to me. @victornoel  WDYT? Is able to merge?

File: src/test/java/org/cactoos/scalar/LengthOfTest.java (hunk 4 above)
Comment (victornoel) [reply src/test/java/org/cactoos/scalar/LengthOfTest.java:221]: @andreoss hehe this was much simpler than what I had in mind :) thx

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by sereshqua** [comment]
@victornoel plz give verdict yourself if my 2nd time is not successful, thanks

**Comment by 0crat** [comment]
> @0crat quality good ([here](https://github.com/yegor256/cactoos/issues/1493#issuecomment-723565168))

@sereshqua There is no quality review for [#1493](https://github.com/yegor256/cactoos/issues/1493), no performer

**Comment by victornoel** [comment]
@sereshqua actually it seems to have worked with a bit of delay ^^ thx for retrying, it seemed to have triggered 0crat :P

**Comment by sereshqua** [comment]
@victornoel "a bit of delay" is the slogan of Zerocracy :D 

Happy New Year!

**Comment by victornoel** [comment]
@sereshqua haha, indeed, luckily this delay is not infinite anymore now they fixed the bot :P happy new year to you too ;)
